web/api/v1/api.go

```diff
@@ -1459,8 +1472,23 @@ func (api *API) rules(r *http.Request) apiFuncResult {
 		return invalidParamError(err, "exclude_alerts")
 	}
 
+	paginationRequest, parseErr := parseListRulesPaginationRequest(r)
+	if parseErr != nil {
+		return invalidParamError(parseErr.err, parseErr.parameter)
+	}
+
 	rgs := make([]*RuleGroup, 0, len(ruleGroups))
+
+	foundToken := false
+
 	for _, grp := range ruleGroups {
+		if paginationRequest != nil && paginationRequest.NextToken != "" && !foundToken {
```

Comment:
FYI: Code might actually simpler if we don't have a separate struct but `groupNextToken` and `groupLimit` variables 🙈  parse can return two vars or so (:

Reply:
will try to remove it.

Reply:
done. it saved a few lines :)